### PR TITLE
Add documentation for Evaluator and minor fix for ast doc.

### PR DESCRIPTION
**Pull request by am357**
*Description of changes:*
Adds documentation for Evaluator. It also:
- Renames `EvalProjectValue` to `EvalSelectValue` to better conform to PartiQL core naming.
- Fixes `partiql-ast` documentation.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jpschorr — APPROVED** [review]

File: partiql-eval/src/eval.rs [hunk 1]
```diff
@@ -199,14 +210,9 @@ impl Evaluable for EvalScan {
...
-pub enum EvalJoinKind {
-    Inner,
-    Left,
-    Right,
-    Full,
-}
-
+/// Represents an evaluation `Join` operator; `Join` joins the tuples from its LHS and RHS based on a logic defined
+/// by [`EvalJoinKind`]. For semantics of PartiQL joins and their distinction with SQL's see sections
+/// 5.3 – 5.7 of [PartiQL Specification — 2007](https://partiql.org/assets/PartiQL-Specification.pdf).
```
Comment (alancai98) [inline partiql-eval/src/eval.rs:215]: What does the "2007" mean?

File: partiql-eval/src/eval.rs [hunk 2]
```diff
@@ -466,6 +486,49 @@ impl Evaluable for EvalFilter {
...
+            false => Some(Value::Bag(Box::new(Bag::from(value)))),
+        }
+    }
+
+    fn update_input(&mut self, input: Value, _branch_num: u8) {
+        self.input = Some(input);
+    }
+}
+
+/// Represents an evaluation `Project` operator; for a given bag of input binding tuples as input
```
Comment (alancai98) [inline partiql-eval/src/eval.rs:528]: Do we want to keep the `SELECT` , `SELECT VALUE`, `SELECT *` naming consistent? Currently it's modeled as `EvalProject`, `EvalSelectValue`, and `EvalProjectAll`. Wouldn't it be more consistent for them to all be:
- `EvalProject`, `EvalProjectValue`, and `EvalProjectAll`
- or `EvalSelect`, `EvalSelectValue`, and `EvalSelectAll`

File: partiql-eval/src/eval.rs [hunk 3]
```diff
@@ -880,6 +923,7 @@ impl EvalExpr for EvalUnaryOpExpr {
     }
 }
 
+/// Represents a PartiQL evaluation `IS` operator, e.g. `IS` in `a IS INT`.
```
Comment (alancai98) [inline partiql-eval/src/eval.rs:926]: Think this technically models the whole expression `a IS INT`, not just the `IS` part

File: partiql-eval/src/eval.rs [hunk 4]
```diff
@@ -898,6 +942,14 @@ impl EvalExpr for EvalIsTypeExpr {
     }
 }
 
+/// Represents an evaluation binary operator, e.g. `+` in `a + b`.
```
Comment (alancai98) [inline partiql-eval/src/eval.rs:945]: This technically models the whole expression, `a + b` and not just the `+`.

File: partiql-eval/src/eval.rs [hunk 5]
```diff
@@ -1018,6 +1070,7 @@ impl EvalExpr for EvalBinOpExpr {
     }
 }
 
+/// Represents an evaluation PartiQL `BETWEEN` operator, e.g. in `x BETWEEN 10 AND 20`.
```
Comment (alancai98) [inline partiql-eval/src/eval.rs:1073]: This expr models the whole between expression (i.e. `x BETWEEN 10 AND 20`)

```suggestion
/// Represents an evaluation PartiQL `BETWEEN` operator, e.g. `x BETWEEN 10 AND 20`.
```

**Review by alancai98 — COMMENTED** [review]
Thanks for adding some missing documentation! Left some minor comments and a question about the rename of `EvalSelectValue`

File: partiql-eval/src/eval.rs (hunk 1 above)
Comment (am357) [reply partiql-eval/src/eval.rs]: Was supposed to be the spec year but not sure where `2007` has come from. The idea is to cater for future spec versions.
Fixed in the new commit.

File: partiql-eval/src/eval.rs (hunk 2 above)
Comment (am357) [reply partiql-eval/src/eval.rs:528]: Not strong opinions, did it this way b/c thought `SELECT VALUE` is PartiQL specific but I changed the names to be more consistent in the new commit.

**Review by alancai98 — APPROVED** [review]
Looks good. Build fails due to a formatting issue caught by `rustfmt`

File: partiql-eval/src/eval.rs [hunk 6]
```diff
@@ -740,6 +780,9 @@ impl EvalExpr for EvalPath {
     }
 }
 
+
```
Comment (jpschorr) [inline partiql-eval/src/eval.rs:783]:
```suggestion
```

**Review by alancai98 — APPROVED** [review]

**Pull request merged**